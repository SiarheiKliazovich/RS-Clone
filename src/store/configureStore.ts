/* eslint-disable @typescript-eslint/ban-ts-comment */
import { applyMiddleware, createStore } from "redux";
import thunkMiddleware from "redux-thunk";

import loggerMiddleware from "./logger";
import rootReducer from "../reducers";

const configureStore = () => {
  const middlewares = [loggerMiddleware, thunkMiddleware];
  const middlewareEnhancer = applyMiddleware(...middlewares);

  const store = createStore(rootReducer, middlewareEnhancer);

  // @ts-ignore:next-line
  if (process.env.NODE_ENV !== "production" && module.hot) {
    // @ts-ignore:next-line
    module.hot.accept("./../reducers", () => store.replaceReducer(rootReducer));
  }

  return store;
};

export default configureStore;
