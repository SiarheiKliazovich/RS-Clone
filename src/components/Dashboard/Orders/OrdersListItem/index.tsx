import { FunctionComponent, useState } from "react";
import { useSelector } from "react-redux";

import { lightTheme, nightTheme } from "../../../../data/constants";
import { IStore } from "../../../../interfaces/store";
import { IOrdersListItem } from "@/src/interfaces/ordersListItem";

const OrdersListItem: FunctionComponent<IOrdersListItem> = ({
  item,
  i,
  deleteItem,
  updateItem,
}) => {
  const appInterfaceStore = useSelector((state: IStore) => state.appInterface);
  const { isNightMode } = appInterfaceStore;

  const [selectedValue, setSelectedValue] = useState("EMPTY");

  const backgroundColor = isNightMode
    ? nightTheme.background.element
    : lightTheme.background.element;

  return (
    <>
      <tr>
        <th>{i + 1}</th>
        <th>{item.date.slice(0, 10)}</th>
        <th>{item.price}</th>
        <th>{item.status}</th>
        <th>
          <button
            style={{ backgroundColor }}
            className="icon"
            onClick={() => deleteItem(item._id)}
          ></button>
        </th>
        <th className="table-select">
          <select
            className="select"
            value={selectedValue}
            onChange={(e) => setSelectedValue(e.target.value)}
          >
            <option value="EMPTY" disabled hidden></option>
            <option value="NON-PAID">NON-PAID</option>
            <option value="PAID">PAID</option>
            <option value="DECLINED">DECLINED</option>
            <option value="IN-PROGRESS">IN PROGRESS</option>
            <option value="FINISHED">FINISHED</option>
          </select>
        </th>
        <th>
          <button
            className="button-apply"
            onClick={() => updateItem(item._id, selectedValue)}
          >
            Apply
          </button>
        </th>
      </tr>
    </>
  );
};

export default OrdersListItem;
