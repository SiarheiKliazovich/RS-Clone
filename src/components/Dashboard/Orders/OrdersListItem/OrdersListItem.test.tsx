import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import OrdersListItem from "../OrdersListItem";

const mockStore = configureStore([]);

describe("OrdersListItem", () => {
  // eslint-disable-next-line
  let store: any;

  beforeEach(() => {
    store = mockStore({
      appInterface: {
        accentColor: {
          static: "#000",
          hover: "#fff",
        },
      },
    });
  });

  const item = {
    _id: "1",
    date: "2022-02-24T12:00:00Z",
    price: 100,
    status: "NON-PAID",
    image: "image",
    userId: "userId",
    sellerId: "sellerId",
  };
  const i = 0;
  const deleteItem = jest.fn();
  const updateItem = jest.fn();

  test("renders correctly", () => {
    const { container } = render(
      <table>
        <tbody>
          <Provider store={store}>
            <OrdersListItem
              item={item}
              i={i}
              deleteItem={deleteItem}
              updateItem={updateItem}
            />
          </Provider>
        </tbody>
      </table>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
