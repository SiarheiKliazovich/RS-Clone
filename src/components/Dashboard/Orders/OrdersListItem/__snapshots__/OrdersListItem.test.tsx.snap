// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrdersListItem renders correctly 1`] = `
<table>
  <tbody>
    <tr>
      <th>
        1
      </th>
      <th>
        2022-02-24
      </th>
      <th>
        100
      </th>
      <th>
        NON-PAID
      </th>
      <th>
        <button
          class="button__delete-order"
        >
          <i
            class="fa fa-remove"
          />
          ordersTable.delete
        </button>
      </th>
      <th
        class="table-select"
      >
        <select
          class="select"
          style="color: rgb(43, 52, 59);"
        >
          <option
            disabled=""
            hidden=""
            value="EMPTY"
          >
            select.change
          </option>
          <option
            value="NON-PAID"
          >
            select.nonPaid
          </option>
          <option
            value="PAID"
          >
            select.paid
          </option>
          <option
            value="DECLINED"
          >
            select.declined
          </option>
          <option
            value="IN-PROGRESS"
          >
            select.inProgress
          </option>
          <option
            value="FINISHED"
          >
            select.finished
          </option>
        </select>
      </th>
      <th>
        <button
          class="button-apply"
          style="background-color: rgb(0, 0, 0);"
        >
          ordersTable.apply
        </button>
      </th>
    </tr>
  </tbody>
</table>
`;
